cmake/utils.cmake

```diff
@@ -14,7 +14,7 @@ function(check_init_submodule path)
         endif()
 
         message(STATUS "Setting up dependencies as git submodules")
-        execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
```

Comment:
Should we keep this --recursive?

Reply:
Probably yeah. I'm not sure if the submodules we have right now have submodule themselves but chances are that future submodules might so let's just keep it recursive unless there's an explicit reason to change ^^